### PR TITLE
Beta

### PR DESCRIPTION
**Pull request by einsteinreloaded**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,7 +26,9 @@
   "homepage": "https://github.com/einsteinreloaded/StoneAge#readme",
   "dependencies": {
     "bluebird": "^3.5.0",
+    "body-parser": "^1.17.2",
     "express": "^4.15.4",
+    "nodemon": "^1.11.0",
```
Comment (khubo) [inline package.json]: move it to *devDependencies*

File: src/client/index.js [hunk 2]
```diff
@@ -1 +1,28 @@
 console.log('this is working!!')
+window.submitUsername=function(){
```
Comment (khubo) [inline src/client/index.js]: dude :-1: , why are u manually appending to window object? It just gives global scope anyways.

File: src/index.js [hunk 3]
```diff
@@ -8,6 +9,13 @@ const app = express()
 // serve the website
 app.use('/', express.static(path.join(__dirname, 'public')))
 
+app.use(bodyParser.json()); // for parsing application/json
+
 app.listen(PORT, () => {
   winston.info(`Server started listening on ${PORT}`)
 })
+
+app.post('/user',function(request,response){
```
Comment (khubo) [inline src/index.js]: move it above  app.listen .

File: src/index.js [hunk 4]
```diff
@@ -8,6 +9,13 @@ const app = express()
 // serve the website
 app.use('/', express.static(path.join(__dirname, 'public')))
 
+app.use(bodyParser.json()); // for parsing application/json
```
Comment (khubo) [inline src/index.js]: use the middlewares first.

File: src/public/bundle.js [hunk 5]
```diff
@@ -1,4 +1,30 @@
 (function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require=="function"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);var f=new Error("Cannot find module '"+o+"'");throw f.code="MODULE_NOT_FOUND",f}var l=n[o]={exports:{}};t[o][0].call(l.exports,function(e){var n=t[o][1][e];return s(n?n:e)},l,l.exports,e,t,n,r)}return n[o].exports}var i=typeof require=="function"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s})({1:[function(require,module,exports){
 console.log('this is working!!')
+window.submitUsername=function(){
```
Comment (khubo) [inline src/public/bundle.js]: not again ! :heart:  :dagger:

File: src/public/index.html [hunk 6]
```diff
@@ -3,7 +3,15 @@
...
 </head>
 <body>
-  Hello world!
+  <div class="header">
+    <h2>StoneAge</h2>
+  </div>
+  <div class="main-container">
+    <div class="userform">
+      <input id="username" type="text" placeholder="Username"/>
+      <button onclick="submitUsername()">Submit</button>
```
Comment (khubo) [inline src/public/index.html]: I think this is the bug , my friend

File: src/client/index.js [hunk 7]
```diff
@@ -1 +1,28 @@
...
+    var username=document.getElementById("username").value;
+    var sessionId;
+    UserAction(username).then(setSession).catch(handleError);
+}
+function UserAction(username) {
```
Comment (khubo) [inline src/client/index.js]: Always use Spaces (2 of 'em , boi)

File: src/index.js (hunk 3 above)
Comment (khubo) [inline src/index.js]: i thought we were using arrow function :dancer:

File: src/client/index.js [hunk 8]
```diff
@@ -1 +1,28 @@
...
+        response => response.text() 
+    )
+}   
+
+function handleError(){
+    alert("Game Initialisation Failed!! Please try again");
+
+}
+
+function setSession(data){
```
Comment (khubo) [inline src/client/index.js]: obviously u havent added *Standard* linter to your editor

File: package.json (hunk 1 above)
Comment (einsteinreloaded) [reply package.json]: ya i forgot that one i was trying to put it globally oops :D

File: src/client/index.js (hunk 2 above)
Comment (einsteinreloaded) [reply src/client/index.js]: actually its not in global scope. I tried without appending to the window.

File: src/client/index.js (hunk 8 above)
Comment (einsteinreloaded) [reply src/client/index.js]: then why did it allow me to commit

**Pull request closed without merging**